crates/hir-ty/src/tests/coercion.rs

```diff
@@ -536,17 +536,15 @@ fn test() {
 
 #[test]
 fn coerce_unsize_generic() {
-    check(
+    check_no_mismatches(
         r#"
 //- minicore: coerce_unsized
-struct Foo<T> { t: T };
-struct Bar<T>(Foo<T>);
+struct Foo<T: ?Sized> { t: T };
+struct Bar<T: ?Sized>(Foo<T>);
 
 fn test() {
     let _: &Foo<[usize]> = &Foo { t: [1, 2, 3] };
-                         //^^^^^^^^^^^^^^^^^^^^^ expected &'? Foo<[usize]>, got &'? Foo<[i32; 3]>
```

Comment:
This test case is effectively the same with the added one, just different kind of coerce unsized